src/tbots.py

```diff
@@ -0,0 +1,96 @@
+#!/opt/tbotspython/bin/python3.8
+
+import os
+import sys
+import iterfzf
+from subprocess import PIPE, run, check_call
+import argparse
+from thefuzz import fuzz
+from thefuzz import process
+
+# thefuzz is a fuzzy string matcher in python
+# https://github.com/seatgeek/thefuzz
+#
+# It returns a match ratio between the input and the choices
+# This is an experimentally determined threshold that works
+# for our bazel commands
+THEFUZZ_MATCH_RATIO_THRESHOLD = 50
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description="Run stuff")
+
+    parser.add_argument("action", choices=["build", "run", "test"])
+    parser.add_argument("search_query")
+    parser.add_argument("-p", "--print_command", action="store_true")
+    parser.add_argument("-d", "--debug_build", action="store_true")
+
+    # These are shortcut args for commonly used arguments on our tests
+    # and full_system. All other arguments are passed through as-is
+    # to the underlying binary/test that is being run (unknown_args)
+    parser.add_argument("-v", "--enable_visualizer", action="store_true")
+    parser.add_argument("-s", "--stop_ai_on_start", action="store_true")
+    args, unknown_args = parser.parse_known_args()
+
+    bazel_queries = {
+        "test": ["bazel", "query", "tests(//...)"],
+        "run": ["bazel", "query", "kind(.*_binary,//...)"],
+        "build": ["bazel", "query", "kind(.*_library,//...)"],
+    }
+
+    # Run the appropriate bazel query and ask thefuzz to find the best matching
+    # target, gauranteed to return 1 result because we set limit=1
+    targets = run(bazel_queries[args.action], stdout=PIPE).stdout.split(b"\n")
+    target, confidence = process.extract(args.search_query, targets, limit=1)[0]
+    target = str(target, encoding="utf-8")
```

Comment:
I think we should have a way to select from the tests that exceed the confidence threshold

for instance: I ran ./tbots.py test penalty and I want to be able to select from

//software/ai/hl/stp/tactic/penalty_kick:penalty_kick_fsm_test
//software/ai/hl/stp/tactic/penalty_kick:penalty_kick_tactic_test
//software/ai/hl/stp/play:penalty_kick_enemy_play_test
//software/ai/hl/stp/play:penalty_kick_play_test

Reply:
interesting, so launch FZF for all tests matching confidence over some value?
or just do `./tbots.py test penaltyfsm` and `./tbots.py test penaltyenemy`

the way the wuzz matches it looks like they can give you n matches with decreasing match ratios, we can just do 1 and click the  "I'm feeling lucky" button equivalent and run it

or we can launch FZF everytime (I personally prefer having the "I'm feeling lucky" with more chars in the search query over having to select from fzf after a shorter search query)

thoughts?

Reply:
how about we can default to I'm feeling lucky and then maybe --list or -l flag to launch fzf

Reply:
Another option would be to have "I'm feeling lucky" as the default, but if the top 3 results (can be different) have a confidence score with in 10 (also can be different) of eachother, we launch fzf? 
In the case of a common query like "test penalty" it could potentially make it clearer that there are multiple similar options, and in the case of other more specific queries with one main target, it would just run the top command.

Reply:
added -i interactive flag

re: checking the confidence score, I tried to look at the values they don't seem to make too much sense like that, there are matches within 5 points that are not what we would be looking for

I've just changed it to trigger interactive mode if the confidence is below the threshold

Reply:
looks great